src/lib/snapper/snapshot.rb

```diff
@@ -0,0 +1,96 @@
+# encoding: utf-8
+
+# ------------------------------------------------------------------------------
+# Copyright (c) 2015 SUSE LLC. All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it under
+# the terms of version 2 of the GNU General Public License as published by the
+# Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, contact Novell, Inc.
+#
+# To contact Novell about this file by physical or electronic mail, you may find
+# current contact information at www.novell.com.
+# ------------------------------------------------------------------------------
+
+require "yast"
+require_relative "snapshot_dbus"
+
+module Yast2
+  module Snapper
+    class Snapshot
+      
+      include Yast::Logger
+
+      attr_accessor :number, :type, :pre_number, :timestamp, :user, :description, :cleanup_algo, :user_data, :number, :config, :strategy
```

Comment:
what is a point of having accessor instead of reader when you cannot save such values? BTW documentation of elements would be nice.